drivers/tee/optee/optee_smc.h

```diff
@@ -321,6 +321,13 @@
 #define OPTEE_SMC_EXCHANGE_CAPABILITIES \
 	OPTEE_SMC_FAST_CALL_VAL(OPTEE_SMC_FUNCID_EXCHANGE_CAPABILITIES)
 
+struct optee_smc_exchange_capabilities_result {
+	unsigned long status;
```

Comment:
I hope we don't need extra checks here and for `optee_smc_call_get_os_revision_result` structure
note: `long` can have 32 or 64 bits

Reply:
Long is always 32bit on ARM and OPTEE is ARM-only. And it is also used int arm_smccc_res struct.

Reply:
what about long on Zephyr side?

Reply:
By "ARM" I meant both arm32 and arm64. But now after further investigation, I am no longer sure about the same size of the types. Anyway, this is not relevant because the type of fields in optee_smc_exchange_capabilities_result and arm_smccc_res are the same anyway, so there can't be a mismatch between them (optee assigns values directly to a0, a1, a2, etc).

Reply:
This is not portable anyways. As I said above - you should not use unions, so this part can be dropped at all.